SnapkitPractice/SnapkitPractice/ViewController.swift

```diff
@@ -0,0 +1,141 @@
+        
+        // 제약 설정
+        makeUI()
+        setAction()
+    }
+    
+    func makeUI() {
+        view.backgroundColor = .white
+        
+        let stackView = UIStackView(arrangedSubviews: [mainTextField, addButton])
+        stackView.axis = .horizontal
+        
+        mainTextField.placeholder = "목록에 추가할 아이템을 작성해주세요"
+        
+        addButton.setTitle("추가", for: .normal)
+        addButton.setTitleColor(.white, for: .normal)
+        addButton.backgroundColor = .link
+        
+        view.addSubview(stackView)
+        view.addSubview(mainTableView)
+        
+        stackView.snp.makeConstraints {
+            $0.height.equalTo(50)
+            $0.leading.equalTo(view.safeAreaLayoutGuide.snp.leading).offset(10)
+            $0.trailing.equalTo(view.safeAreaLayoutGuide.snp.trailing).offset(-10)
+            $0.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(10)
+        }
+        
+        mainTextField.snp.makeConstraints {
+            $0.leading.equalTo(stackView.snp.leading).offset(10)
+            $0.centerY.equalTo(stackView.snp.centerY)
+            $0.top.equalTo(stackView.snp.top)
+            $0.bottom.equalTo(stackView.snp.bottom)
+        }
+        
+        addButton.snp.makeConstraints {
+            $0.width.equalTo(50)
+            $0.leading.equalTo(mainTextField.snp.trailing)
+            $0.trailing.equalTo(stackView.snp.trailing).offset(-10)
+            $0.top.equalTo(stackView.snp.top)
+            $0.bottom.equalTo(stackView.snp.bottom)
+        }
+        
+        mainTableView.snp.makeConstraints {
+            $0.top.equalTo(stackView.snp.bottom).offset(10)
+            $0.bottom.equalTo(view.safeAreaLayoutGuide.snp.bottom)
+            $0.leading.equalTo(view.safeAreaLayoutGuide.snp.leading)
+            $0.trailing.equalTo(view.safeAreaLayoutGuide.snp.trailing)
+        }
+    }
+    
+    func setAction() {
+        addButton.addTarget(self, action: #selector(addButtonClicked), for: .touchUpInside)
+    }
+    
+    @objc func addButtonClicked() {
+        let studentName = mainTextField.text!
+        
+        guard let studentFirstName = studentName.first else { return }
+        StudentManager.shared.append(firstName: studentFirstName, fullName: studentName)
+        
+        mainTextField.text = ""
+        mainTableView.reloadData()
+    }
+}
+
+extension ViewController: UITableViewDelegate, UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return StudentManager.shared.sectiontList.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        let firstName = FirstName(rawValue: section)
+        
+        switch firstName {
+        case .kim:  return "김씨 리스트"
+        case .park: return "박씨 리스트"
+        case .choi: return "최씨 리스트"
+        case .lee:  return "이씨 리스트"
+        case .none: return nil
```

Comment:
김이박최에 속하지 않는 사람은 "기타" 섹션에 담아주세요~!

```suggestion
        default: return "기타"
```

Reply:
수정 완료했습니다!